anndata/compat/__init__.py

```diff
@@ -230,6 +229,8 @@ def _find_sparse_matrices(d: Mapping, n: int, keys: tuple, paths: list):
 
 
 def _slice_uns_sparse_matrices(uns: MutableMapping, oidx: "Index1d", orig_n_obs: int):
+    from anndata._core.index import _subset
```

Comment:
moving this inside method did the job. We can think of better ways to handle the optional dependency afterwards